test/e2e/framework/test_context.go

```diff
@@ -526,6 +530,14 @@ func AfterReadingAllFlags(t *TestContextType) {
 	}
 }
 
+const (
+	// This is the traditional gomega.Format default of 4000 for an object
+	// dump plus some extra room for the message.
+	maxFailureMessageSize = 5000
```

Comment:
/hold

Let's first discuss in https://github.com/kubernetes/test-infra/pull/27831#discussion_r1008446740 what a good maximum size would be.

Reply:
[Here's what we use in triage](https://github.com/kubernetes/test-infra/blob/2c20bdb10a0e3177a1163cbb2b4f83b450d7df70/triage/summarize/summarize.go#L31-L32)

10K when clustering things together, 100K when displaying failure text (IIRC this came about from integration tests having a ton of output)

So I'm fine with at least 10K, I feel like this is something we can adjust as followup if we don't like it

Reply:
That's significantly larger than what we produce here, so 5000 should be safe. Let's merge and see how well it works in practice.

/hold cancel